spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkReadConf.java

```diff
@@ -83,7 +91,14 @@ public Long endSnapshotId() {
   }
 
   public String branch() {
-    return confParser.stringConf().option(SparkReadOptions.BRANCH).parseOptional();
+    String optionBranch = confParser.stringConf().option(SparkReadOptions.BRANCH).parseOptional();
```

Comment:
Do we plan to support the read option too?

Reply:
Discussed with Ryan offline a bit, the consensus we had was we can support it, but it should be discouraged. Let us know what you think!

Reply:
What's the use for the option if we support AS OF tag/branch? My worry is that we add something that is going to work only in some cases.

Reply:
The commit https://github.com/apache/iceberg/commit/305e320b0fc8fb7aebc447fc291777c27fb8486b was a part of 1.1 release, so it will cause compatibility issue.

Reply:
Okay, then it makes sense to keep it. Resolving.